packages/nextjs/utils/scaffold-eth/networks.ts

```diff
@@ -54,40 +54,31 @@ export const NETWORKS_EXTRA_DATA: Record<string, TChainAttributes> = {
  * @param txnHash
  * @dev returns empty string if the network is localChain
  */
-export const getBlockExplorerTxLink = (network: Network, txnHash: string) => {
-  const { name, chainId } = network;
+export function getBlockExplorerTxLink(network: Network, txnHash: string) {
+  let blockExplorerTxURL = "";
+  const { chainId } = network;
 
-  if (chainId === 31337 || chainId === 1337) {
-    // If its localChain then return empty sting
-    return "";
-  }
+  const chainNames = Object.keys(chains);
 
-  let blockExplorerNetwork = "";
-  if (name && chainId > 1) {
-    blockExplorerNetwork = name + ".";
-  }
+  const targetChainArr = chainNames.filter(chainName => {
+    const wagmiChain = chains[chainName as keyof typeof chains];
+    return wagmiChain.id === chainId;
+  });
 
-  let blockExplorerBaseTxUrl = "https://" + blockExplorerNetwork + "etherscan.io/tx/";
-  if (chainId === 100) {
-    blockExplorerBaseTxUrl = "https://blockscout.com/poa/xdai/tx/";
+  if (targetChainArr.length === 0) {
+    return "";
   }
 
-  const blockExplorerTxURL = blockExplorerBaseTxUrl + txnHash;
-
-  return blockExplorerTxURL;
-};
+  const targetChain = targetChainArr[0] as keyof typeof chains;
+  // @ts-expect-error : ignoring error since `blockExplorers` key may or may not be present and we are already checking in ternary
+  blockExplorerTxURL = chains[targetChain]?.blockExplorers?.default?.url
+    ? // @ts-expect-error
+      chains[targetChain].blockExplorers.default.url
+    : "";
```

Comment:
I'll love learn why this happens.

If I do `targetChain = "goerli"`, everything works without the `@ts-expect-error`.

I get that the `blockExplorers` might not be defined for some chains (since is optional in `type Chain`), but doing this before:

```js
if (!chains[targetChain].blockExplorers) {
  return "";
}
```

should work, right? (or the check in the ternary also)

I'm guessing it looks statically to all the possible chain options, and if one of them (e.g hardhat) doesn't have the prop, it throws the `Property 'blockExplorers' does not exist on type` error.

Not sure if this is the expected behavior or if we are missing something. Nothing important, just curious :D 

cc @sverps @rin-st

Reply:
Yeah I believe your guess is correct. It can be simplified though (since we check the value for truthyness a bit lower anyway):

```typescript
// @ts-expect-error : ignoring error since `blockExplorers` key may or may not be present and we are already checking in ternary
blockExplorerTxURL = chains[targetChain]?.blockExplorers?.default?.url;
```

Or alternatively the typecasting solution (which doesn't require any ts-ignore):
```typescript
type ChainWithExplorer = Extract<typeof chains[typeof targetChain], { blockExplorers: unknown }>;
blockExplorerTxURL = (chains[targetChain] as ChainWithExplorer)?.blockExplorers?.default?.url;
```

Reply:
Love the TS ninja stuff with Extract :D 

But went with the first option a0ac6136936df5fef9e517fbea670036f937dc30